### ISSUE TITLE
Run CI on PRs and master

### ISSUE THREAD

**Issue by mikkelbu** [nunit/nunit-console#1386]
It seems like we are only running two checks on PRs: WIP and license/cla. So this means that neither AppVeyor nor Azure pipelines are being run as checks on PRs or master (note that AppVeyor run on on both PR and master, but not as a check). 

![image](https://github.com/nunit/nunit-console/assets/1007631/e977271e-6df2-4569-80ef-95161e52fb70)

**Comment by CharliePoole** [comment]
@mikkelbu I turned off Azure back when I was working on it a year ago. It wasn't working well for me and I was unfamiliar with Azure - and didn't care to get familiar. :-) Working as a contributor, I think I'd just go do something else if I had azure failures right now - AppVeyor is bad enough!

I've always been an advocate of asking committers to not commit until everything is passing but I'm frankly not a fan of software-enforced rules. For one thing, it smells like distrust to me and you shouldn't give people commit authority if you don't trust them to follow sensible rules on their own. Also, most rules need to be overridden from time to time and that often requires a higher level of authority - i.e. asking @OsirisTerje to do it. Given that so few people are working on this, it seems wrong to add obstacles.

As it happens, I just had to push directly to main because of an unresolved. GitVersion bug encountered when trying to run a PR. If that check were in place, I would have been stuck, but I could see that the error had nothing to do with what I was doing and was able to merge and test locally, then push and trigger a build of main, which resolved the bug.
